### PR TITLE
Issue14795 aggregator test

### PR DESCRIPTION
**Pull request by zycz**

### CODE REVIEW

File: doc/nrf/releases/release-notes-changelog.rst [hunk 1]
```diff
@@ -254,6 +254,8 @@ Common Application Framework (CAF)
   The :kconfig:option:`CONFIG_CAF_BLE_STATE_SECURITY_REQ` can be used to enable this feature.
   The option can be used for both Bluetooth Peripheral and Bluetooth Central.
 
+* :ref:`caf_sensor_data_aggregator` unit tests added.
+
 |no_changes_yet_note|
```
Comment (greg-fer) [inline doc/nrf/releases/release-notes-changelog.rst:259]: Remove this line -- there are already changes, so it is not needed.

File: doc/nrf/releases/release-notes-changelog.rst [hunk 2]
```diff
@@ -254,6 +254,8 @@ Common Application Framework (CAF)
   The :kconfig:option:`CONFIG_CAF_BLE_STATE_SECURITY_REQ` can be used to enable this feature.
   The option can be used for both Bluetooth Peripheral and Bluetooth Central.
 
+* :ref:`caf_sensor_data_aggregator` unit tests added.
```
Comment (greg-fer) [inline doc/nrf/releases/release-notes-changelog.rst:257]:
```suggestion
* :ref:`caf_sensor_data_aggregator`:

  * Added unit tests for the library.
```

**Review by greg-fer — CHANGES_REQUESTED** [review]

**Comment by gopiotr** [comment]
Please add also register into CODEOWNERS file with path to test `/tests/subsys/caf/sensor_data_aggregator/` and person/people responsible for this test:
https://github.com/nrfconnect/sdk-nrf/blob/main/CODEOWNERS

File: tests/subsys/caf/sensor_data_aggregator/src/main.c [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+	for (; i > 0; i--) {
+		struct sensor_event *se = new_sensor_event(sizeof(struct sensor_value) *
+				ORDER_TEST_SENSOR_SAMPLE_SIZE);
+
+		zassert_not_null(se, "Failed to allocate event");
+		se->descr = ORDER_TEST_AGG_DESCR;
+		se->dyndata.size = sizeof(struct sensor_value) * ORDER_TEST_SENSOR_SAMPLE_SIZE;
+		se->dyndata.data[0] = i;
+		APP_EVENT_SUBMIT(se);
+		k_yield();
```
Comment (emob-nordic) [inline tests/subsys/caf/sensor_data_aggregator/src/main.c:91]: This is something i wouldn't expect in the tests. If you are lacking of the HEAP just increase it. This looks like something is not good within the test.
Furthermore if the test is named `test_order` i would expect we are firstly flood the system with events and then we check if they came in correct order. With this yield you would probably create and receive event one by one. It kinda "tests" what event manager ensures - event order.

File: tests/subsys/caf/sensor_data_aggregator/src/main.c [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+#include "test_config.h"
+#include <zephyr/drivers/sensor.h>
+
+static enum test_id cur_test_id;
+static K_SEM_DEFINE(test_end_sem, 0, 1);
+
+
+void test_init(void)
+{
+	zassert_false(app_event_manager_init(), "Error when initializing");
```
Comment (rakons) [inline tests/subsys/caf/sensor_data_aggregator/src/main.c:21]: zassert_ok

File: tests/subsys/caf/sensor_data_aggregator/src/main.c [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+	cur_test_id = test_id;
+	struct test_start_event *ts = new_test_start_event();
+
+	zassert_not_null(ts, "Failed to allocate event");
+	ts->test_id = test_id;
+	APP_EVENT_SUBMIT(ts);
+
+	int err = k_sem_take(&test_end_sem, K_SECONDS(30));
+
+	zassert_equal(err, 0, "Test execution hanged");
```
Comment (rakons) [inline tests/subsys/caf/sensor_data_aggregator/src/main.c:35]: zassert_ok

File: tests/subsys/caf/sensor_data_aggregator/src/main.c [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+		zassert_not_null(se, "Failed to allocate event");
+		se->descr = BASIC_TEST_AGG_DESCR;
+		se->dyndata.size = sizeof(struct sensor_value) * BASIC_TEST_SENSOR_SAMPLE_SIZE;
+		APP_EVENT_SUBMIT(se);
+		k_yield();
+	}
+
+	int err = k_sem_take(&test_end_sem, K_SECONDS(30));
+
+	zassert_equal(err, 0, "Test execution hanged");
```
Comment (rakons) [inline tests/subsys/caf/sensor_data_aggregator/src/main.c:68]: zassert_ok

File: tests/subsys/caf/sensor_data_aggregator/src/main.c [hunk 7]
```diff
@@ -0,0 +1,136 @@
...
+		se->descr = ORDER_TEST_AGG_DESCR;
+		se->dyndata.size = sizeof(struct sensor_value) * ORDER_TEST_SENSOR_SAMPLE_SIZE;
+		se->dyndata.data[0] = i;
+		APP_EVENT_SUBMIT(se);
+		k_yield();
+	}
+
+	int err = k_sem_take(&test_end_sem, K_SECONDS(30));
+
+	zassert_equal(err, 0, "Test execution hanged");
```
Comment (rakons) [inline tests/subsys/caf/sensor_data_aggregator/src/main.c:96]: zassert_ok

File: tests/subsys/caf/sensor_data_aggregator/src/main.c [hunk 8]
```diff
@@ -0,0 +1,136 @@
...
+
+		zassert_equal(cur_test_id, ev->test_id,
+			"End test ID does not equal start test ID");
+		cur_test_id = TEST_IDLE;
+		k_sem_give(&test_end_sem);
+
+		return false;
+	}
+
+	zassert_true(false, "Wrong event type received");
```
Comment (rakons) [inline tests/subsys/caf/sensor_data_aggregator/src/main.c:131]: zassert_unreachable

File: tests/subsys/caf/sensor_data_aggregator/src/modules/test_basic.c [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+			/* Ignore other test cases, check if proper test_id. */
+			zassert_true(st->test_id < TEST_CNT,
+				     "test_id out of range");
+			break;
+		}
+
+		return false;
+	}
+
+	zassert_true(false, "Event unhandled");
```
Comment (rakons) [inline tests/subsys/caf/sensor_data_aggregator/src/modules/test_basic.c:78]: zassert_unreachable

File: tests/subsys/caf/sensor_data_aggregator/src/modules/test_data_receiver.c [hunk 10]
```diff
@@ -0,0 +1,100 @@
...
+			zassert_not_null(te, "Failed to allocate event");
+			te->test_id = cur_test_id;
+			APP_EVENT_SUBMIT(te);
+		}
+
+		return false;
+	}
+
+
+	zassert_true(false, "Event unhandled");
```
Comment (rakons) [inline tests/subsys/caf/sensor_data_aggregator/src/modules/test_data_receiver.c:93]: zassert_unreachable

**Review by rakons — APPROVED** [review]
Only style notes of zasserts usage.

**Review by gopiotr — APPROVED** [review]

**Review by emob-nordic — APPROVED** [review]

**Pull request merged**